### ISSUE TITLE
terminfo.LookupTerminfo segfaults, fails to detect terms with infocmp

### ISSUE THREAD

**Issue by CosmicToast** [zyedidia/tcell#16]
this is related to zyedidia/micro#1854 (I was testing the solution and it wouldn't launch, I got to this point via delve)

tl;dr `if t.TrueColor` will segfault if t is nil
the check for `t == nil` should be moved up further

however, the secondary issue here is that it fails to detect the `foot` terminal, despite infocmp being present/installed (admittedly in `~/.terminfo`, which is standard (see terminfo(5)'s "Fetching Compiled Descriptions")
I couldn't find (in a reasonable timeframe) exactly where infocmp (or anything similar) is invoked, but this is clearly differing behavior (esp. since infocmp is expected to be installed)

**Comment by zyedidia** [comment]
Thanks, I fixed the nil pointer crash. I'm not too familiar with infocmp, but tcell was using it to detect terminfo entries (see `terminfo/dynamic/dynamic.go`) and it seems like it does not find entries in `~/.terminfo` or `/usr/share/terminfo` which is problematic.

I've updated tcell to use a pure go library for detecting and reading terminfo files (better than relying on infocmp anyway) so I believe this should be fixed now. I haven't been able to test with foot, so let me know if it works for you.
